### PR TITLE
spec source and xml updates for cl_intel_device_attribute_query

### PR DESCRIPTION
**Pull request by bashbaug**
This PR adds the spec source and XML updates for the `cl_intel_device_attribute_query` extension.

I will create an OpenCL-Headers PR with updates for this extension shortly.

I will create an OpenCL-Registry PR after the spec source is merged.

### CODE REVIEW

File: extensions/cl_intel_device_attribute_query.asciidoc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        If the implementation is driven primarily by a PCI device with a PCI device ID, the low 16 bits of the device identifier must contain that PCI device ID, and the remaining bits must be set to zero.
+        Otherwise, the choice of what to return may be dictated by operating system or platform policies - but should uniquely identify both the device version and any major configuration options (for example, core count in the case of multi-core devices).
+| {CL_DEVICE_NUM_SLICES_INTEL}
+  | `cl_uint`
+      | The number of slices in the device.
+        A slice is a collection of sub-slices.
+| {CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL}
+  | `cl_uint`
+      | The maximum number of sub-slices in one slice.
+        A slice is a collection of execution units (EUs), fixed-function units, and caches.
```
Comment (jlewis-austin) [inline extensions/cl_intel_device_attribute_query.asciidoc:144]: Slices are defined two different ways, while sub-slices aren't explicitly defined here. Maybe move this definition up to `CL_DEVICE_NUM_SLICE_INTEL` and here talk about what a sub-slice is? I'm not sure if it's relevant or not, but I was curious if the sub-slices are assumed or required to be the equivalent (ie, all contain the same numbers and types of units and caches).

File: extensions/cl_intel_device_attribute_query.asciidoc [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      | The maximum number of threads that may be simultaneously resident on one execution unit (EU).
+| {CL_DEVICE_FEATURE_CAPABILITIES_INTEL}
+  | {cl_device_feature_capabilities_intel_TYPE}
+      | Feature flags describing capabilities supported by the device.
+|====
+--
+
+== Issues
+
+. How to express cache sizes?  Is for the entire device?
```
Comment (jlewis-austin) [inline extensions/cl_intel_device_attribute_query.asciidoc:160]: Second sentence could use an edit. Maybe refer to the scope of the cache(s) being queried?

**Review by jlewis-austin — APPROVED** [review]
LGTM, just a couple editorial ideas

File: extensions/cl_intel_device_attribute_query.asciidoc (hunk 1 above)
Comment (alycm) [reply extensions/cl_intel_device_attribute_query.asciidoc:144]:
```suggestion
        A sub-slice is a collection of execution units (EUs), fixed-function units, and caches.
```
I had a similar thought as @jlewis-austin when reading this.  No idea if my suggestion is correct, but it seems to fit the context.

File: extensions/cl_intel_device_attribute_query.asciidoc [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+Additional feature flags may be added in subsequent versions of the extension.
+
+[source]
+----
+typedef cl_bitfield         cl_device_feature_capabilities_intel;
+
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_FEATURE_FLAG_DP4A_INTEL         (1 << 0)
+#define CL_DEVICE_FEATURE_FLAG_DPAS_INTEL         (1 << 1)
```
Comment (alycm) [inline extensions/cl_intel_device_attribute_query.asciidoc:101]: It is not required for merging, but pointing out that there is no description of these capabilities in this document.

**Review by alycm — APPROVED** [review]
LGTM, neither of my comments are blockers.

File: extensions/cl_intel_device_attribute_query.asciidoc (hunk 2 above)
Comment (bashbaug) [reply extensions/cl_intel_device_attribute_query.asciidoc:160]: Yes, it should be:

```suggestion
. How to express cache sizes?  Is it for the entire device?
```

Thanks!

File: extensions/cl_intel_device_attribute_query.asciidoc (hunk 1 above)
Comment (bashbaug) [reply extensions/cl_intel_device_attribute_query.asciidoc]: Thank you for catching this, that's a little embarassing!

Alastair's suggested fix is correct.  I've applied it.

File: extensions/cl_intel_device_attribute_query.asciidoc (hunk 3 above)
Comment (bashbaug) [reply extensions/cl_intel_device_attribute_query.asciidoc:101]: Yes - we will either document these in an updated version of the spec or in a related implementation doc.  Thanks!

**Pull request merged**